internal/shellgen/tmpl/flake_remove_nixpkgs.nix.tmpl

```diff
@@ -29,25 +37,59 @@
             {{- end }}
             {{- end }}
           ];
+
+          overlays = [
+            {{- range $flake.Packages }}
+            {{- if .PatchGlibc }}
+            (final: prev: {
+              {{.PackageAttributePath}} = prev.{{.PackageAttributePath}};
+            })
+            {{- end }}
+            {{- end }}
+          ];
         });
         {{- end }}
         {{- end }}
+
+	{{ if .PatchGlibc -}}
+        patchGlibc = pkg: derivation {
+          name = "devbox-patched-glibc";
+          system = "{{.System}}";
+
+          # Set these attributes so that glibc-patch.bash has access to their
+          # paths via environment variables of the same name.
+          inherit pkg;
+          glibc = nixpkgs.legacyPackages."{{.System}}".glibc;
```

Comment:
I suggest using `stdenv.cc.libc_lib`.  I don't know if `glibc` nix package is always reliably updated. For example, glibc 2.36 versions are missing https://www.nixhub.io/packages/glibc

Reply:
I specified `glibc` because the patch would break if the stdenv libc was different (for example, it could be musl).